packages/headless/src/features/commerce/pagination/pagination-slice.ts

```diff
@@ -114,5 +114,5 @@ function getEffectiveSlice(
 }
 
 function handlePaginationReset(state: CommercePaginationState) {
-  state.principal = getCommercePaginationInitialSlice();
+  state.principal.page = getCommercePaginationInitialSlice().page;
```

Comment:
About this change: This was causing issues in atomic, when we need to check the "what's the number of products as reported by pagination" to know "are there products available, if there's no product we should hide this section".

It seems wrong to reset to original state; Instead I think we should be only resetting the "current page" to 0, not full pagination state.

Reply:
Seems similar to what is done in the search pagination reducer :+1: https://github.com/coveo/ui-kit/blob/702a04a919cce2df64e4555df5acdc6e787b0bc5/packages/headless/src/features/pagination/pagination-slice.ts#L163